### ISSUE TITLE
Can't pull container uploaded by action by returned digest

### ISSUE THREAD

**Issue by minor-fixes** [docker/build-push-action#419]
#### Steps to reproduce this issue

1. Create a workflow that builds and pushes a container to ghcr.io, and outputs the digest
2. Try to pull this image by digest (e.g. `docker pull my/image@sha256:digest_from_action_output`)

#### Expected behaviour

`docker pull` by digest is successful

#### Actual behaviour

`docker pull` by digest returns:

```
Error response from daemon: manifest unknown
```

### What appears to be happening

Image is built and pushed to ghcr.io successfully. Build reports the digest as e.g. `sha256:058cdcda7e8904028f980c6aca8a9f616ded4862c7d6045db34e3e9ec140dd2b`.

In ghcr.io, this digest corresponds with the `config.digest`: 
![image](https://user-images.githubusercontent.com/8988434/126857463-1e8ec212-e1ae-46d9-a4c8-ae20d7466081.png)

Pulling with this digest doesn't work:

![image](https://user-images.githubusercontent.com/8988434/126857569-0e8b25a4-069a-4c44-94b7-929e1529028f.png)

Pulling with the other digest:

![image](https://user-images.githubusercontent.com/8988434/126857606-0cd8c7a5-b93f-4bda-9032-79e2d6db1172.png)

does work:

![image](https://user-images.githubusercontent.com/8988434/126857586-df2ba3c2-1e18-478f-b9e5-146c12402d31.png)

**TL;DR**: How can I get this action to report the top-level digest which I can use to pull?

---

Some reverse-engineering of the docker command that my configuration corresponds to yields:

```
docker buildx build --push --tag ghcr.io/minorhacks/autobio:master --iidfile /tmp/iidfile .
```

which indeed has a digest in both the console output and the `iidfile`, but they are both of the `config.digest` variant.

A normal `docker push`, however, prints the top-level `digest`, as in:

```
docker build --tag ghcr.io/minorhacks/autobio:master .
docker push ghcr.io/minorhacks/autobio:master
```

which yields a digest that ghcr.io can pull by.

Is there a way to expose this "pullable" digest? I'm looking to propagate this value to other actions which will pull from the registry by that digest.

**Comment by crazy-max** [comment]
@minor-fixes Can you post your workflow and logs please? I can't repro on my side with the following use cases:

```Dockerfile
FROM alpine
RUN echo "Hello world!"
```

```yaml
name: iidfile

on:
  workflow_dispatch:

env:
  DOCKER_IMAGE: localhost:5000/name/app

jobs:
  build:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        include:
          - driver: docker
            load: false
            push: true
            platforms: linux/amd64
          - driver: docker
            load: true
            push: false
            platforms: linux/amd64
          - driver: docker-container
            load: false
            push: true
            platforms: linux/amd64
          - driver: docker-container
            load: true
            push: false
            platforms: linux/amd64
          - driver: docker-container
            load: false
            push: true
            platforms: linux/amd64,linux/arm64
    services:
      registry:
        image: registry:2
        ports:
          - 5000:5000
    steps:
      -
        name: Checkout
        uses: actions/checkout@v2
      -
        name: Set up QEMU
        uses: docker/setup-qemu-action@v1
      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1
        with:
          version: latest
          driver: ${{ matrix.driver }}
          driver-opts: |
            network=host
      -
        name: Build
        id: docker_build
        uses: docker/build-push-action@v2
        with:
          context: .
          file: ./Dockerfile
          load: ${{ matrix.load }}
          push: ${{ matrix.push }}
          tags: ${{ env.DOCKER_IMAGE }}:latest
          platforms: ${{ matrix.platforms }}
      -
        name: Digest
        run: |
          echo "${{ steps.docker_build.outputs.digest }}"
      -
        name: Docker images
        run: |
          docker image ls --no-trunc
      -
        name: Pull
        if: ${{ matrix.push && matrix.driver != 'docker' }}
        run: |
          docker pull ${{ env.DOCKER_IMAGE }}@${{ steps.docker_build.outputs.digest }}
      -
        name: Check manifest
        if: ${{ matrix.push }}
        run: |
          docker buildx imagetools inspect ${{ env.DOCKER_IMAGE }}:latest
      -
        name: Inspect
        if: ${{ matrix.load }}
        run: |
          docker image inspect ${{ env.DOCKER_IMAGE }}:latest
```

**Comment by minor-fixes** [comment]
Apologies for the slow response. In my attempts to work around this, I've updated my runner machine a bunch - maybe this is just a matter of having woefully out-of-date tooling. I'll try to reproduce and upload some logs.
